### PR TITLE
Class based restructurization, support for an additional backend (speech-dispatcher), license changed to GPL

### PR DESCRIPTION
**Pull request by kratuvid**
Plus:
1. New option to use the python piper module (--piper_python)
2. Changed the /stop endpoint to /reset because it is more descriptive, and seems to conflict with /pause
3. More informative /status endpoint

Notes:
1. Incomplete speech dispatcher support. Only playback works for now
2. License changed for the sake of software freedom (https://www.gnu.org/philosophy/free-software-even-more-important.en.html). Permissive licenses are not free (as in speech)

Tell me if you agree with the changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,45 +5,45 @@ Select and read aloud text from anywhere 🔊
...
-2. Create a virtual environment, install requirements and run:
+1. Install piper in your $PATH if using the original C++ variant, or `pip install piper-tts` if using the Python wrapper
+2. Download the models and their respective configurations in a directory. See [here](https://github.com/rhasspy/piper/blob/master/VOICES.md)
+3. Create a virtual environment, install requirements and run:
     ```bash
-    python -m venv venv
+    python -m venv venv --system-site-packages
     source venv/bin/activate
     pip install -r requirements.txt 
     python main.py --port 5000 --playback_speed=1.0 --volume_level=.8 --model yourmodel.onnx --model_config yourmodel.onnx.json --wayland
```
Comment (dipta10) [inline README.md]:
```suggestion
    python main.py --port 5000 --speed=1.0 --volume=.8 --piper-model yourmodel.onnx --piper-model-config yourmodel.onnx.json --wayland
```

File: piper_backend.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+            finally:
+                self.play_process.set(None)
+                self.play_queue.task_done()
+                with self.play_queue_size.lock:
+                    self.play_queue_size.data = self.play_queue_size.data - len(audio)
+                    self.play_queue_size.data = max(0, self.play_queue_size.data)
+
+    def run_gen_thread(self):
+        while True:
+            text, getaudio = self.gen_queue.get()
```
Comment (dipta10) [inline piper_backend.py:92]: We should probably add back the text sanitization, which is really useful in some cases, especially for some PDFs
```py
unidecode(text).replace("‐\n", "").replace("‐ ", "")
```

File: main.py [hunk 3]
```diff
@@ -1,382 +1,233 @@
...
+        help="Piper: Playback sample rate. More info at https://github.com/rhasspy/piper/blob/master/TRAINING.md",
+    )
+    parser.add_argument(
+        "--piper-sentence-silence",
+        type=float,
+        default=0.8,
+        help="Piper: Seconds of silence after each sentence",
+    )
+    parser.add_argument(
+        "--piper-one-sentence",
```
Comment (dipta10) [inline main.py:212]: I think you might have missed implementing the one-sentence part. You could add this if you want or I should be able to add this during the weekend.

**Comment by dipta10** [comment]
Thank you for continuing to contribute to this project. Appreciate it 😁
I didn't know the part about the license, thanks for fixing it!

I was not able to review the codebase thoroughly, but the application is working correctly on my machine. Once you address the merge conflict I will merge it. And later this weekend I will try to review the codebase thoroughly.

**Review by dipta10 — CHANGES_REQUESTED** [review]

File: piper_backend.py (hunk 2 above)
Comment (kratuvid) [reply piper_backend.py:92]: I forgot about that. Done in main.py:read()

File: main.py (hunk 3 above)
Comment (kratuvid) [reply main.py:212]: Yeah. I can do it

**Comment by kratuvid** [comment]
LGTM. Tested with affected configurations. No side effects noticed

**Comment by dipta10** [comment]
hi @kratuvid , I was not able to review the new changes, but since it looks good to you I'll merge it. Thanks for your contribution.

**Review by dipta10 — APPROVED** [review]

**Pull request merged**